packages/core/src/routes/session.ts

```diff
@@ -34,12 +37,27 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
             passwordEncrypted && passwordEncryptionMethod && passwordEncryptionSalt,
             'session.invalid_sign_in_method'
           );
+
           assertThat(
             encryptPassword(id, password, passwordEncryptionSalt, passwordEncryptionMethod) ===
               passwordEncrypted,
-            'session.invalid_credentials'
+            new RequestErrorWithUserLog('session.invalid_credentials', {
+              id: nanoid(),
+              userId: id,
+              type: UserLogType.SignInUsernameAndPassword,
+              result: UserLogResult.Failed,
+              payload: {},
+            })
           );
 
+          await insertUserLog({
+            id: nanoid(),
+            userId: id,
+            type: UserLogType.SignInUsernameAndPassword,
+            result: UserLogResult.Success,
+            payload: {},
+          });
```

Comment:
1. the constructing of these payloads could be extracted into an until since they are too similar.
2. as i wrote above, `insertUserLog` can be put in a middleware for success logs. no need to build a general log middleware now, we can start with sth like `userLogMiddleware`.

Reply:
`userLogMiddleware` added.